Updated help messages for install.sh and rmake.py

rkamd opened the pull request:
resolves # [LWPMLSE-162](https://ontrack-internal.amd.com/browse/LWPMLSE-162)

Summary of proposed changes:
- Updated help messages in install.sh
- Updated help messages in rmake.py

TorreZuk reviewed (commented):
Some suggestions to consider ,  can see what others say.

daineAMD commented on install.sh:
```diff
@@ -8,28 +8,65 @@
+    -i | --install                Pass this flag to generate and install library package after build.
+
+    -d | --dependencies           Pass this flag to also build and install external dependencies.
+                                  Dependecies are to be installed in /usr/local. This should be done only once.
```
Yes, see https://github.com/ROCmSoftwarePlatform/hipBLAS/blob/develop/install.sh#L441

rkamd replied:
Fixed the typo

rkamd commented on install.sh:
```diff
@@ -8,28 +8,65 @@
+  Options:
+    -h | --help                   Print this help message.
+
+    -i | --install                Pass this flag to generate and install library package after build.
```
I agree but my preferrence is to keep it as "Pass this flag", for the following reasons
- In hipcc "Pass" phrase is used for options taking arguments such as "Pass <arg>". Here, the options do no take any args, so I think it is fine to explicity mention to the users that these are flags. 
- It matches the rocSOLVER help message
@amcamd / @daineAMD,
Let me know your preference, I will update rocBLAS install script accordingly.

amcamd replied:
My preference would be to drop the "Pass this flag" when it is only a flag, and use "Specify" when it is a flag + a string, so:
- Print this help message
- Generate ans install library package after build
- Build and install external dependencies. Dependencies are to be installed in /usr/local. This should be done only once.
- ...
- Specify path to an existing rocBLAS install directory (eg. /home/rocBLAS/build/release/rocblas-install)
- Specify rocBLAS version (eg. 2.42.0)
- Specify path to an existing rocSOLVER install directory (eg. /home/rocSOLVER/build/release/rocsolver-install)

amcamd commented on install.sh:
```diff
@@ -8,28 +8,65 @@
+                                  (e.g. /src/rocBLAS/build/release/rocblas-install)
+
+    -b | --rocblas                Pass this flag to specify rocblas version.
+
```
should this be 
-b | --rocblas <version>

rkamd replied:
Yes, I will fix it

amcamd commented on install.sh:
```diff
@@ -8,28 +8,65 @@
+    --custom-target <target>      Pass this flag to link the library against custom target (eg. host, device).
+
+    -v | --rocm-dev <version>     Pass this flag to set specific rocm-dev version.
+
```
I think this should be:
Specify rocm-dev version (eg. 4.5.0)

amcamd commented on install.sh:
```diff
@@ -8,28 +8,65 @@
+    --rocsolver-path <solverdir>  Specify path to an existing rocSOLVER install directory.
+                                  (e.g. /src/rocSOLVER/build/release/rocsolver-install)
+
+    --compiler </compier/path>    Specify path to host compiler.
```
can you add an (eg. /opt/rocm/bin)  you will need to test that the example works

rkamd commented:
> I think this should be:
>Specify rocm-dev version (eg. 4.5.0) 

Agree

amcamd commented on install.sh:
```diff
@@ -8,28 +8,65 @@
+    -p | --cmakepp                Pass this flag for to add CMAKE_PREFIX_PATH
+
+    --custom-target <target>      Pass this flag to link the library against custom target (eg. host, device).
+
```
I think this should be:
Specify custom target to link the library against (eg. host, device)  

You will need to check that the examples work

rkamd commented:
Shall I sort the options in alphabetical order? @amcamd,@torrezuk,@daineAMD

rkamd commented:
In the last commit, I have addressed the Andrew's review comments and also re-arranged the options in sorted order. In the install.sh script I replaced '|' symbol to a comma which matches the format used by linux commands.

amcamd reviewed (approved):
This is a nice improvement, should make install.sh more user friendly.

amcamd commented:
There is more information on --custom-target here: https://github.com/ROCm-Developer-Tools/HIP/issues/2158

There is more information on --rocm-dev here: https://github.com/ROCmSoftwarePlatform/rocBLAS/pull/1040

Both of the above options are for expert users, the first specifies the target device for the compiler. The second specifies an install base for the rocm dev-package
